ls1intum/Artemis
# `Development`: Modernize appearance of course, exercise and user statistics

### Checklist
#### General


#### Client
### Motivation and Context


This PR switches the implementations for the chart in course and exercise statistics from `ng2-charts`/ `chart.js` to `ngx-charts`.
### Description

The HTML templates as well as the underlying logic has been adapted in order to fit` ngx-charts`' requirements.
I additionally introduced a new style file and util. These are not directly necessary now but it allows me to remove some code duplicates and hopefully unifies the styling for vertical bar charts. 
Last but not least, I centralized the three doughnut charts in the exercise statistics because I think it is better this way.
### Steps for Testing

Prerequisites:
- 1 Administrator
- 1 Course with multiple exercises and some participation
- 1 Exercise with some participation

**Important**: As always, I recommend to look at the old implementation first before deploying this PR. Take some screenshots, this might make it easier to compare both implementations later on

1. Log in to Artemis
2. Navigate to Course Administration
3. Click on the course card and then on "Statistics"
4. Inspect the charts, change the time span, check tooltips and translations
5. Go to the exercise overview page and click on the title of the corresponding exercise
6. Again, click on "Statistics" and inspect the charts. Test the behavior for different button clicks
7. In the "Server administration" menu, select "User statistics"
8. Again, inspect the charts. 

Comments:

- As you might observe, the line in the average scores chart for the course statistics is missing in the new implementation. Instead, the average course score is written on the left. This is the solution @krusche and I discussed, as such bar-line combinations are not supported by `ngx-charts` per default.

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage






### Screenshots


Old course statistics:
![image](https://user-images.githubusercontent.com/80622272/146575426-ac710664-1f42-4cd2-a284-c1ab2b99c4ff.png)

**New** Course statistics:
![image](https://user-images.githubusercontent.com/80622272/146576345-513cad47-0a47-439b-8f47-d2258b946b3f.png)

Old Exercise statistics:
![image](https://user-images.githubusercontent.com/80622272/146575708-858c311d-0be7-45e3-8c3c-65935cf90569.png)

**New** Exercise statistics:
![image](https://user-images.githubusercontent.com/80622272/146578395-63fdea31-be5d-4d19-a64d-6a4f6e7ae797.png)


Old User statistics:
![image](https://user-images.githubusercontent.com/80622272/146575798-9b176192-0b43-4fc7-adda-19ab65bb0ed0.png)

**New** User statistics:
![image](https://user-images.githubusercontent.com/80622272/146578671-ec75765e-ef91-4d88-9373-b7a79387d62c.png)
